java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java

```diff
@@ -88,61 +89,29 @@
         Job.getJobManager().join(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, monitor);
 
         final ICompilationUnit unit = JDTUtils.resolveCompilationUnit(uri);
-        // whether the file is on test path or not is guarded at client side cache
-        if (!isJavaElementExist(unit) || monitor.isCanceled()) {
+        final IType primaryType = unit.findPrimaryType();
+        if (!isJavaElementExist(unit) || primaryType == null || monitor.isCanceled()) {
             return resultList;
         }
 
-        final IType[] childrenTypes = unit.getAllTypes();
-        final Map<String, TestItem> classMapping = new HashMap<>();
-        for (final IType type : childrenTypes) {
-            if (!isTestableClass(type)) {
-                continue;
-            }
-            final List<TestItem> testMethodList = Arrays.stream(type.getMethods()).map(method -> {
-                try {
-                    return TestFrameworkUtils.resoveTestItemForMethod(method);
-                } catch (final JavaModelException e) {
-                    return null;
-                }
-            }).filter(Objects::nonNull).collect(Collectors.toList());
-            TestItem classItem = null;
-            if (testMethodList.size() > 0) {
-                classItem = TestItemUtils.constructTestItem(type, TestLevel.CLASS);
-                for (final TestItem method : testMethodList) {
-                    resultList.add(method);
-                    classItem.addChild(method.getId());
-                }
-                // Assume the kinds of all methods are the same.
-                classItem.setKind(testMethodList.get(0).getKind());
-                resultList.add(classItem);
-            } else {
-                // JUnit 5 supports nested test classes
-                if (isJunit5TestableClass(type)) {
-                    classItem = TestItemUtils.constructTestItem(type, TestLevel.CLASS, TestKind.JUnit5);
-                    resultList.add(classItem);
-                }
+        final ASTParser parser = ASTParser.newParser(AST.JLS14);
+        parser.setSource(unit);
+        parser.setFocalPosition(0);
+        parser.setResolveBindings(true);
```

Comment:
Consider you only need check the method signature and it's annotation, it's appropriate to ignore the method body during parsing the AST. This probably boost the performance.

`parser.setIgnoreMethodBodies(true);`